src/support/name.h

```diff
@@ -33,34 +31,40 @@ namespace wasm {
 // TODO: as an optimization, IString values < some threshold could be considered
 //       numerical indices directly.
 
-struct Name : public cashew::IString {
-  Name() : cashew::IString() {}
-  Name(const char* str) : cashew::IString(str, false) {}
-  Name(cashew::IString str) : cashew::IString(str) {}
-  Name(const std::string& str) : cashew::IString(str.c_str(), false) {}
+struct Name : public IString {
+  Name() : IString() {}
+  Name(std::string_view str) : IString(str, false) {}
+  Name(const char* str) : IString(str, false) {}
+  Name(IString str) : IString(str) {}
+  Name(const std::string& str) : IString(str) {}
+
+  // String literals do not need to be copied. Note: Not safe to construct from
+  // temporary char arrays! Take their address first.
+  template<size_t N> Name(const char (&str)[N]) : IString(str) {}
```

Comment:
I guess I could add a non-const overload as well. Since I've literally never seen `char const foo[] = {'f', 'o', 'o', '\0'}` in the wild (note the placement of the `const`), this should make it almost perfectly safe in practice.

I can also add some unit tests that make sure this behaves as intended under ASan.

Reply:
Sounds good.

lgtm overall with benchmarking, though there is an ASan error on CI now and I'd like to see what the fix for that is.

Reply:
Yes, I looked now and there are a few places it is used (because we have `Name("<some string literal")`, but not as many as I had expected.

Reply:
Aren't they using
```cpp
Name(const char* str) : IString(str, false) {} 
```
?

By the way I commented out this constructor and the code still compiles.

Reply:
It turns out that the type of a string literal is `const char[N]`, not `const char*`, so if this worked as intended, they would end up in this constructor. However, it turns out that this is not working correctly due to weird function resolution rules described here: https://stackoverflow.com/questions/16708307/is-it-possible-to-legally-overload-a-string-literal-and-const-char. The fix would require making the `const char*` constructor a template, too, and that doesn't seem worth it. I'll just remove this attempt at optimizing string literals for now.